packages/polka/index.js

```diff
@@ -61,10 +61,11 @@ class Polka extends Router {
 	}
 
 	handler(req, res, next) {
-		let info = this.parse(req, true);
+		let info = this.parse(req);
 		let obj = this.find(req.method, req.path=info.pathname);
 
-		req.params = obj.params;
+		req.params = {};
+		Object.entries(obj.params).forEach(([key, val]) => (req.params[key] = decodeURIComponent(val)));
```

Comment:
You should really get out of the habit of `Object.entries` – it's an abysmal performer: https://esbench.com/bench/6116a3726c89f600a5701026

 
```suggestion
		req.params = {};
		for (let key in obj.params) {
		  req.params[key] = decodeURIComponent(obj.params[key]);
		}
```